storage/pebble/open.go

```diff
@@ -0,0 +1,106 @@
+package pebble
+
+import (
+	"fmt"
+
+	"errors"
+
+	"github.com/cockroachdb/pebble"
+	"github.com/hashicorp/go-multierror"
+
+	"github.com/onflow/flow-go/storage"
+	"github.com/onflow/flow-go/storage/pebble/registers"
+)
+
+// NewBootstrappedRegistersWithPath initializes a new Registers instance with a pebble db
+// if the database is not initialized, it close the database and return pebble.ErrNotBootstrapped
+func NewBootstrappedRegistersWithPath(dir string) (*Registers, *pebble.DB, error) {
+	db, err := OpenRegisterPebbleDB(dir)
+	if err != nil {
+		return nil, nil, fmt.Errorf("failed to initialize pebble db: %w", err)
+	}
+	registers, err := NewRegisters(db)
+	if err != nil {
+		if errors.Is(err, ErrNotBootstrapped) {
+			// closing the db if not bootstrapped
+			dbErr := db.Close()
+			if dbErr != nil {
+				err = multierror.Append(err, fmt.Errorf("failed to close db: %w", dbErr))
+			}
+		}
+		return nil, nil, fmt.Errorf("failed to initialize registers: %w", err)
+	}
+	return registers, db, nil
+}
+
+// openRegisterPebbleDB opens the database
+func OpenRegisterPebbleDB(dir string) (*pebble.DB, error) {
```

Comment:
```suggestion
func OpenRegisterDB(dir string) (*pebble.DB, error) {
```
nit: We can drop Pebble since the package is pebble I guess.